### DISCUSSION TITLE
how to sync history only when ctrl+r pressed?

### DISCUSSION THREAD

**Discussion by taikulawo** [fish-shell/fish-shell#10545] [Q&A]
I want to sync history across session. there is a similar issue
https://github.com/fish-shell/fish-shell/issues/825#issuecomment-440286038

my different shell session has different command, so I don't want press `up` sync history from another fish shell. I only want to sync when I press ctrl +r 

I have tried something but not works. could you give me a idea?
```fish
bind \cx "history --merge; set -l x (commandline -b); commandline -r ''; commandline -r \$x"
```

**Comment by taikulawo** [comment]
fish `fzf` already support this feature
https://github.com/PatrickF1/fzf.fish/blob/8920367cf85eee5218cc25a11e209d46e2591e7a/functions/_fzf_search_history.fish#L5

**Reply by krobelus** [reply]
`bind \cx "history merge"` kinda works but only after pressing `enter` (on the empty commandline).
We could probably fix that somehow
